src/modules/registration/providers/registration/registration.ts

```diff
@@ -0,0 +1,67 @@
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { Print } from '@print';
+import { Registration } from '../../classes/registration';
+
+// tslint:disable-next-line:no-unused no-unused-variable
+const LOG_REF = '[RegistrationService]';
+
+const RESOURCE_PATH = '/registrations';
+
+@Injectable()
+export class RegistrationService {
+
+  // Indicates wether or not the user has already registered (this reset at each app launch).
+  userRegistered: boolean;
+  // The registraiton object saved on the backend (this reset at each app launch).
+  registration: Registration;
+
+  constructor(private readonly http: HttpClient) {
+    this.userRegistered = false;
+    this.registration = {};
```

Comment:
No need to change anything in this PR, but for future reference, I would prefer to store this kind of state as an observable in the future.

Storing it like it's done here requires us to manually sync this service's state and the states of the components that use it.

Using a `BehaviorSubject` and registering to this service's state as an observable frees us from having to do it, and allows us to react to future state change with no further code.